src/arkode/arkode_io.c

```diff
@@ -2072,6 +2072,41 @@ int ARKodeResetAccumulatedError(void* arkode_mem)
   /* Reset value and counter, and return */
   ark_mem->AccumErrorStart = ark_mem->tn;
   ark_mem->AccumError      = ZERO;
+
+  return (ARK_SUCCESS);
+}
+
+int ARKodeSetAdjointCheckpointScheme(void* arkode_mem,
+                                     SUNAdjointCheckpointScheme checkpoint_scheme)
+
+{
+  ARKodeMem ark_mem;
+  if (arkode_mem == NULL)
+  {
+    arkProcessError(NULL, ARK_MEM_NULL, __LINE__, __func__, __FILE__,
+                    MSG_ARK_NO_MEM);
+    return (ARK_MEM_NULL);
+  }
+  ark_mem = (ARKodeMem)arkode_mem;
+
+  ark_mem->checkpoint_scheme = checkpoint_scheme;
+
+  return (ARK_SUCCESS);
+}
+
+int ARKodeSetAdjointCheckpointIndex(void* arkode_mem, int64_t step_index)
```

Comment:
```suggestion
int ARKodeSetAdjointCheckpointIndex(void* arkode_mem, suncountertype step_index)
```

Reply:
I think I've noted most of them but it would be good to grep for `int64_t` to make sure we've not missed any `int64_t` to `suncountertype` changes